_posts/2021-03-10-katib-0.10.md

```diff
@@ -0,0 +1,310 @@
+the Kubernetes cluster. To make this modification, it is necessary to know:
+
+- what API group, version, and kind the Kubernetes CRD has, and
+
+- which Kubernetes resources the CRD's controller creates.
+
+Check the
+[Kubernetes guide](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/)
+to know more about CRDs.
+
+Follow these two simple steps to integrate new CRD in Katib:
+
+1. Modify
+   the Katib controller [Deployment's arguments](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-controller.yaml#L26)
+   with the new flag:
+
+   ```
+   --trial-resources=<object-kind>.<object-API-version>.<object-API-group>
+   ```
+
+   For example, to support Tekton **Pipeline**:
+
+   ```yaml
+   . . .
+   containers:
+     - name: katib-controller
+       image: docker.io/kubeflowkatib/katib-controller
+       command: ["./katib-controller"]
+       args:
+         - "--webhook-port=8443"
+         - "--trial-resources=Job.v1.batch"
+         - "--trial-resources=TFJob.v1.kubeflow.org"
+         - "--trial-resources=PyTorchJob.v1.kubeflow.org"
+         - "--trial-resources=MPIJob.v1.kubeflow.org"
+         - "--trial-resources=PipelineRun.v1beta1.tekton.dev"
+    . . .
+   ```
+
+1. Modify the Katib controller
+   [ClusterRole's rules](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/rbac.yaml#L5)
+   with the new rule to give Katib an access to all Kubernetes resources that
+   are created by the CRD's controller. To know more about ClusterRole,
+   please check the [Kubernetes guide](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#role-and-clusterrole).
+
+   For example, for the Tekton **Pipeline**, Trial creates Tekton **PipelineRun**,
+   then Tekton **PipelineRun** creates Tekton **TaskRun**.
+   Therefore, Katib controller ClusterRole should have an access to the _pipelineruns_ and _taskruns_:
+
+   ```yaml
+   kind: ClusterRole
+   apiVersion: rbac.authorization.k8s.io/v1
+   metadata:
+     name: katib-controller
+   rules:
+     - apiGroups:
+         - tekton.dev
+       resources:
+         - pipelineruns
+         - taskruns
+       verbs:
+         - "*"
+     - apiGroups:
+         - kubeflow.org
+       resources:
+         - tfjobs
+         - pytorchjobs
+         - mpijobs
+       verbs:
+         - "*"
+   . . .
+   ```
+
+1. Install Katib by following
+   [the getting started guide](https://www.kubeflow.org/docs/components/katib/hyperparameter/#installing-katib).
+
+At this point, the Kubernetes CRD can be used in the Katib Trial template. Check
+[this guide](https://github.com/kubeflow/katib/tree/master/examples/v1beta1/tekton)
+to know more about Tekton and Katib integration.
+
+## Early Stopping
+
+Early Stopping is now supported in the 0.10 release of Katib.
+Early Stopping is one of the essential steps for doing HP tuning. It helps to
```

Comment:
Since this is the first mention of HP tuning, can we rephrase to "hyper-parameter (HP) tuning"?

Reply:
I think the first mention of HP was here: https://github.com/kubeflow/blog/blame/32ae07c1db42d680d1b843f0c414143da18c351c/_posts/2021-03-10-katib-0.10.md#L30.